File name too long

Running suckit on https://uncyclopedia.kr eventually fails with an error like this:
```
2024-12-02 15:06:45.030494981 +09:00: [ERROR] Couldn't create folder uncyclopedia.kr/wiki/%EB%B0%B1%EA%B4%B4%EB%89%B4%EC%8A%A4%ED%86%A0%EB%A1%A0:%EB%82%98%EB%AE%88%EC%97%90%ED%8A%B8_%EC%97%B0%EB%B0%A9%EA%B3%B5%ED%99%94%EA%B5%AD_%ED%8C%8C1%EB%A9%B8%ED%95%98%EB%8B%A4%3F_%E3%85%88%EB%AC%B8%EA%B0%80_%EB%A6%AC%EB%8F%99%EC%9D%B4%EB%A5%BC_%EC%A7%81%EC%A0%91_%EB%AA%A8%EC%85%94_%EC%99%94%EC%8A%B5%EB%8B%88%EB%8B%A4!: File name too long (os error 36)
```
I believe this could be related with #8

I am facing the same issue : 
```console
2025-06-12 20:22:59.990273242 +02:00: [ERROR] Couldn't create folder ./strategie-prod-renew.ext.ssl-gouv.fr/debats__querystring__keys=&term_node_tid_depth_join=8441&term_node_tid_depth_join_1=All&field_date_de_publication_value%5bvalue%5d%5byear%5d=&field_date_de_publication_value_1%5bvalue%5d%5bmonth%5d=&field_report_display_value=All&field_date_evenement_value=All: File name too long (os error 36)

thread '<unnamed>' panicked at src/logger.rs:42:9:
Couldn't create folder ./strategie-prod-renew.ext.ssl-gouv.fr/debats__querystring__keys=&term_node_tid_depth_join=8441&term_node_tid_depth_join_1=All&field_date_de_publication_value%5bvalue%5d%5byear%5d=&field_date_de_publication_value_1%5bvalue%5d%5bmonth%5d=&field_report_display_value=All&field_date_evenement_value=All: File name too long (os error 36)

thread '<unnamed>' panicked at src/scraper.rs:180:57:
called `Result::unwrap()` on an `Err` value: PoisonError { .. }

thread '<unnamed>' panicked at src/scraper.rs:180:57:
called `Result::unwrap()` on an `Err` value: PoisonError { .. }

thread 'main' panicked at src/scraper.rs:328:10:
called `Result::unwrap()` on an `Err` value: Any { .. }
```

--- a/src/scraper.rs
+++ b/src/scraper.rs
@@ -192,7 +192,10 @@ impl Scraper {
 
                 let next_full_url = match url.join(url_to_parse.as_str()) {
                     Ok(url) => url,
-                    Err(e) => panic!("Failed to parse url: {} | Error: {}", next_url, e),
+                    Err(e) => {
+                        warn!("Failed to parse url: {} | Error: {}", next_url, e);
+                        return;
+                    },
                 };
 
                 let path = url_helper::to_path(&next_full_url, true);

--- a/src/url_helper.rs
+++ b/src/url_helper.rs
@@ -1,6 +1,7 @@
 use std::path::Path;
 
 use md5;
+use std::borrow::Cow;
 use url::Url;
 
 ///Max file name size supported by the file system
@@ -24,6 +25,19 @@ pub fn to_path(url: &Url, with_fragment: bool) -> String {
         .parent()
         .map_or("", |filename| filename.to_str().unwrap())
         .to_string();
+
+    // Ensure the folder names are not too long
+    parent = parent
+        .split('/')
+        .map(|str| {
+            if str.len() > FILE_NAME_MAX_LENGTH {
+                Cow::Owned(format!("{:x}", md5::compute(str)))
+            } else {
+                Cow::Borrowed(str)
+            }
+        })
+        .collect::<Vec<Cow<str>>>()
+        .join("/");
 
     if url_path_and_query.ends_with('/') {
         filename = "index.html".to_string();

